packages/components/src/date-time/date.js

```diff
@@ -1,24 +1,130 @@
 /**
  * External dependencies
  */
-import ReactDatePicker from 'react-datepicker';
+import classnames from 'classnames';
 import moment from 'moment';
+import { DayPickerSingleDateController } from 'react-dates';
+
+/**
+ * WordPress dependencies
+ */
+import { withInstanceId } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
 
 /**
  * Module Constants
  */
 const TIMEZONELESS_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
 
-function DatePicker( { currentDate, onChange, ...args } ) {
-	const momentDate = currentDate ? moment( currentDate ) : moment();
-	const onChangeMoment = ( newDate ) => onChange( newDate.format( TIMEZONELESS_FORMAT ) );
+class DatePicker extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			descriptionVisible: false,
+		};
+
+		this.onClickDescriptionToggle = this.onClickDescriptionToggle.bind( this );
+	}
+
+	onClickDescriptionToggle() {
+		this.setState( { descriptionVisible: ! this.state.descriptionVisible } );
+	}
+
+	render() {
+		const { currentDate, onChange, instanceId } = this.props;
+
+		const momentDate = currentDate ? moment( currentDate ) : moment();
+		const momentTime = {
+			seconds: momentDate.seconds(), minutes: momentDate.minutes(),
+			hours: momentDate.hours(),
+		};
+
+		const onChangeMoment = ( newDate ) => {
+			onChange( newDate.set( momentTime ).format( TIMEZONELESS_FORMAT ) );
+		};
 
-	return <ReactDatePicker
-		inline
-		selected={ momentDate }
-		onChange={ onChangeMoment }
-		{ ...args }
-	/>;
+		return (
+			<div className="components-datetime-date">
+				<DayPickerSingleDateController
+					aria-describedby={ `datepicker-container__aria-description-${ instanceId }` }
```

Comment:
Seems to me this `aria-describedby` is not rendered. Whether or not `DayPickerSingleDateController ` supports it, I'd say we don't want all the wall of text in the Help to be announced. I'd recommend to remove it.